packages/ui/src/components/OpenApi/OpenApiForm.tsx

```diff
@@ -133,27 +144,24 @@ export const OpenApiForm: FunctionComponent = () => {
           const entity = new CamelRouteVisualEntity(route);
           entitiesContext?.camelResource.addExistingEntity(entity);
         }
-      }
-    }
-
-    const rest: Rest = {
-      openApi: {
-        specification: downloadUrl,
-      },
-    };
+      });
+    });
 
+    const rest: Rest = FlowTemplateService.getFlowTemplate(SourceSchemaType.Rest);
+    rest.openApi = rest.openApi ?? { specification: '' };
+    rest.openApi.specification = downloadUrl;
     let restExists: boolean = false;
 
     entitiesContext?.camelResource.getVisualEntities().forEach((entity: BaseVisualCamelEntity) => {
       if (entity.type === EntityType.Rest) {
-        if ((entity as CamelRestVisualEntity).rest.rest?.openApi?.specification === downloadUrl) {
-            restExists = true;
+        if ((entity as CamelRestVisualEntity).restDef.rest?.openApi?.specification === downloadUrl) {
+          restExists = true;
         }
       }
     });
 
     if (!restExists) {
-      const entity = new CamelRestVisualEntity(rest);
+      const entity = new CamelRestVisualEntity({ rest });
```

Comment:
The constructor of the `CamelRestVisualEntity` class is:

```typescript
constructor(public restDef: { rest: Rest }) {
```
but we were passing `rest: Rest` directly